design/control-data-plane-separation/design.md

```diff
@@ -23,377 +23,118 @@ architecture does not allow the control plane and data plane to scale independen
+  - We could introduce a CRD, but where would it attach? We already have NginxProxy which controls dynamic data plane configuration, and this may eventually attach to the Gateway instead of just the GatewayClass. Would a Deployment configuration fit in there, and would it be dynamic? That would require us to completely redeploy nginx if a user changes those settings.
+  - We could start with the helm chart option, and rely on user feedback to see if we need to get more granular.
+  - This could also involve creating a ConfigMap that the control plane consumes on startup and contains all nginx Deployment/Daemonset configuration, including NGINX Plus usage configuration.
+- nginx Service should have configurable labels and annotations via the GatewayInfrastructure field in the Gateway resource.
+- Control plane creates the nginx deployment and service when a Gateway resource is created, in the same namespace as the Gateway resource. When the Gateway is deleted, the control plane deletes nginx deployment and service.
+- Control plane should label the nginx service and deployment with something related to the name of the Gateway so it can easily be linked.
+- Liveness/Readiness probes:
+  - Control plane probe currently waits until we configure nginx. Going forward, this probe should just be when the control plane is ready to configure.
+  - nginx probe...can agent expose any type of health endpoint? This is not yet clear.
+- Control plane should not restart data plane pods if they are unhealthy.
 
-The user will deploy the control plane, and data plane via Kubernetes manifests. The control plane supports a
-single `GatewayClass` and  `Gateway` resource. Following this model, it makes sense for the control plane to support a
-single data plane Deployment/DaemonSet. Each agent+nginx Pod in the data plane will receive the same configuration from
-the control plane. In the future, we may reconsider this model if there’s a use case for the control plane to support
-multiple `Gateway` resources. For example, in that model, it might make sense for the control plane to be responsible
-for provisioning a separate data plane for each `Gateway.`
+### Agent Configuration
 
-### Running Multiple NGF Controllers
+Using [nginx-agent.conf](https://github.com/nginx/agent/blob/v3/nginx-agent.conf), we can configure the agent on startup. Note that this example conf file may not have all available options. At a minimum, it will need to be configured for the following:
 
-Users can run multiple NGF controllers in the same cluster by leveraging the `GatewayClass` resource. Multiple NGF
-controllers will be able to coexist in the same cluster as long as they each have a unique corresponding `GatewayClass`
-resource. In this case, each installation of NGF will contain a unique `GatewayClass` resource, a control plane
-Deployment, and a data plane Deployment/DaemonSet.
+- command server is the NGF ClusterIP service
+  - tls settings for this connection
```

Comment:
Will the user have to bring their own cert/key to secure this connection?

Reply:
Yes, I can link to the encryption section below